qualcommbe: ipq95xx: Add support for SPI-NAND, PCIe, USB3 and general fixup

Ansuel opened the pull request:
Followup of https://github.com/openwrt/openwrt/pull/17725

Enable and add support for SPI-NAND, USB3, PCIe and general fixup.

What has been tested and works correctly:
- USB3 (attached USB drive and read/write file)
- SPI-NAND (correctly read the rootfs partition with hexdump and ubi could attach correctly/write)

We also add some documentation about the RPM problem. I hope Vendor board doesn't suffer from this.

Anyway the new boot log is the following. With ping and iperf result as asked. @wigyori
```
Jumping to AARCH64 kernel via monitor
[    0.000000] Booting Linux on physical CPU 0x0000000000 [0x411fd090]
[    0.000000] Linux version 6.6.71 (ansuel@Ansuel-XPS24) (aarch64-openwrt-linux-musl-gcc (OpenWrt GCC 13.3.0 r0+27995-57f2c72505) 13.3.0, GNU ld (GNU Binutils) 2.42) #0 SMP Wed Jan 29 20:35:02 2025
[    0.000000] Machine model: Qualcomm Technologies, Inc. IPQ9574/AP-AL02-C7
[    0.000000] OF: reserved mem: 0x000000004a100000..0x000000004a4fffff (4096 KiB) nomap non-reusable bootloader@4a100000
[    0.000000] OF: reserved mem: 0x000000004a500000..0x000000004a5fffff (1024 KiB) nomap non-reusable sbl@4a500000
[    0.000000] OF: reserved mem: 0x000000004a600000..0x000000004a9fffff (4096 KiB) nomap non-reusable tz@4a600000
[    0.000000] OF: reserved mem: 0x000000004aa00000..0x000000004aafffff (1024 KiB) nomap non-reusable smem@4aa00000
[    0.000000] Zone ranges:
[    0.000000]   DMA      [mem 0x0000000040000000-0x000000007fffffff]
[    0.000000]   DMA32    empty
[    0.000000]   Normal   empty
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000040000000-0x000000004a0fffff]
[    0.000000]   node   0: [mem 0x000000004a100000-0x000000004aafffff]
[    0.000000]   node   0: [mem 0x000000004ab00000-0x000000007fffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000040000000-0x000000007fffffff]
[    0.000000] psci: probing for conduit method from DT.
[    0.000000] psci: PSCIv1.0 detected in firmware.
[    0.000000] psci: Using standard PSCI v0.2 function IDs
[    0.000000] psci: MIGRATE_INFO_TYPE not supported.
[    0.000000] psci: SMC Calling Convention v1.0
[    0.000000] percpu: Embedded 18 pages/cpu s35624 r8192 d29912 u73728
[    0.000000] Detected VIPT I-cache on CPU0
[    0.000000] CPU features: detected: Spectre-v2
[    0.000000] CPU features: detected: Spectre-v4
[    0.000000] CPU features: detected: Spectre-BHB
[    0.000000] alternatives: applying boot alternatives
[    0.000000] Kernel command line: console=ttyMSM0,115200n8 ubi.mtd=rootfs root=mtd:ubi_rootfs rootfstype=squashfs
[    0.000000] Dentry cache hash table entries: 131072 (order: 8, 1048576 bytes, linear)
[    0.000000] Inode-cache hash table entries: 65536 (order: 7, 524288 bytes, linear)
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 258048
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] software IO TLB: SWIOTLB bounce buffer size adjusted to 1MB
[    0.000000] software IO TLB: area num 4.
[    0.000000] software IO TLB: mapped [mem 0x000000007ea80000-0x000000007eb80000] (1MB)
[    0.000000] Memory: 996648K/1048576K available (8448K kernel code, 944K rwdata, 2704K rodata, 7616K init, 279K bss, 51928K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=4, Nodes=1
[    0.000000] rcu: Hierarchical RCU implementation.
[    0.000000]  Tracing variant of Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.
[    0.000000] NR_IRQS: 64, nr_irqs: 64, preallocated irqs: 0
[    0.000000] Root IRQ handler: gic_handle_irq
[    0.000000] GICv2m: range[mem 0x0b00c000-0x0b00cffc], SPI[640:671]
[    0.000000] GICv2m: range[mem 0x0b00d000-0x0b00dffc], SPI[672:703]
[    0.000000] GICv2m: range[mem 0x0b00e000-0x0b00effc], SPI[704:735]
[    0.000000] rcu: srcu_init: Setting srcu_struct sizes based on contention.
[    0.000000] arch_timer: cp15 and mmio timer(s) running at 24.00MHz (virt/virt).
[    0.000000] clocksource: arch_sys_counter: mask: 0xffffffffffffff max_cycles: 0x588fe9dc0, max_idle_ns: 440795202592 ns
[    0.000000] sched_clock: 56 bits at 24MHz, resolution 41ns, wraps every 4398046511097ns
[    0.000047] Calibrating delay loop (skipped), value calculated using timer frequency.. 48.00 BogoMIPS (lpj=240000)
[    0.000056] pid_max: default: 32768 minimum: 301
[    0.003990] Mount-cache hash table entries: 2048 (order: 2, 16384 bytes, linear)
[    0.004001] Mountpoint-cache hash table entries: 2048 (order: 2, 16384 bytes, linear)
[    0.007052] RCU Tasks Trace: Setting shift to 2 and lim to 1 rcu_task_cb_adjust=1 rcu_task_cpu_ids=4.
[    0.007179] rcu: Hierarchical SRCU implementation.
[    0.007182] rcu:     Max phase no-delay instances is 1000.
[    0.007630] smp: Bringing up secondary CPUs ...
[    0.009074] Detected VIPT I-cache on CPU1
[    0.009161] CPU1: Booted secondary processor 0x0000000001 [0x411fd090]
[    0.010766] Detected VIPT I-cache on CPU2
[    0.010851] CPU2: Booted secondary processor 0x0000000002 [0x411fd090]
[    0.012451] Detected VIPT I-cache on CPU3
[    0.012535] CPU3: Booted secondary processor 0x0000000003 [0x411fd090]
[    0.012620] smp: Brought up 1 node, 4 CPUs
[    0.012627] SMP: Total of 4 processors activated.
[    0.012631] CPU features: detected: 32-bit EL0 Support
[    0.012635] CPU features: detected: CRC32 instructions
[    0.012687] CPU features: emulated: Privileged Access Never (PAN) using TTBR0_EL1 switching
[    0.012692] CPU: All CPU(s) started at EL1
[    0.012694] alternatives: applying system-wide alternatives
[    0.017752] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.017765] futex hash table entries: 1024 (order: 4, 65536 bytes, linear)
[    0.019200] pinctrl core: initialized pinctrl subsystem
[    0.020707] NET: Registered PF_NETLINK/PF_ROUTE protocol family
[    0.021449] DMA: preallocated 128 KiB GFP_KERNEL pool for atomic allocations
[    0.021505] DMA: preallocated 128 KiB GFP_KERNEL|GFP_DMA pool for atomic allocations
[    0.021557] DMA: preallocated 128 KiB GFP_KERNEL|GFP_DMA32 pool for atomic allocations
[    0.021809] thermal_sys: Registered thermal governor 'step_wise'
[    0.021846] cpuidle: using governor menu
[    0.021970] ASID allocator initialised with 65536 entries
[    0.025835] /soc@0/interrupt-controller@b000000: Fixed dependency cycle(s) with /soc@0/interrupt-controller@b000000
[    0.037948] Modules: 27728 pages in range for non-PLT usage
[    0.037956] Modules: 519248 pages in range for PLT usage
[    0.042055] SCSI subsystem initialized
[    0.042174] usbcore: registered new interface driver usbfs
[    0.042214] usbcore: registered new interface driver hub
[    0.042239] usbcore: registered new device driver usb
[    0.042943] qcom_scm: convention: smc arm 64
[    0.043937] s1: Bringing 0uV into 725000-725000uV
[    0.044098] qcom_rpm_smd_regulator remoteproc:glink-edge:rpm-requests:regulators: Supply for s1 (s1) resolved to itself
[    0.044189] l5: Bringing 0uV into 1800000-1800000uV
[    0.044743] clocksource: Switched to clocksource arch_sys_counter
[    0.047791] NET: Registered PF_INET protocol family
[    0.047893] IP idents hash table entries: 16384 (order: 5, 131072 bytes, linear)
[    0.050019] tcp_listen_portaddr_hash hash table entries: 512 (order: 1, 8192 bytes, linear)
[    0.050035] Table-perturb hash table entries: 65536 (order: 6, 262144 bytes, linear)
[    0.050046] TCP established hash table entries: 8192 (order: 4, 65536 bytes, linear)
[    0.050096] TCP bind hash table entries: 8192 (order: 6, 262144 bytes, linear)
[    0.050325] TCP: Hash tables configured (established 8192 bind 8192)
[    0.050717] MPTCP token hash table entries: 1024 (order: 2, 24576 bytes, linear)
[    0.050863] UDP hash table entries: 512 (order: 2, 16384 bytes, linear)
[    0.050890] UDP-Lite hash table entries: 512 (order: 2, 16384 bytes, linear)
[    0.051242] NET: Registered PF_UNIX/PF_LOCAL protocol family
[    0.051273] PCI: CLS 0 bytes, default 64
[    0.056218] workingset: timestamp_bits=46 max_order=18 bucket_order=0
[    0.056835] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.056840] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.064008] qcom-qusb2-phy 7b000.phy: Registered Qcom-QUSB2 phy
[    0.066008] qcom-pcie 10000000.pcie: host bridge /soc@0/pcie@10000000 ranges:
[    0.066080] qcom-pcie 10000000.pcie:       IO 0x0010200000..0x00102fffff -> 0x0000000000
[    0.066100] qcom-pcie 10000000.pcie:      MEM 0x0010300000..0x0017ffffff -> 0x0010300000
[    0.066110] qcom-pcie 18000000.pcie: host bridge /soc@0/pcie@18000000 ranges:
[    0.066141] qcom-pcie 18000000.pcie:       IO 0x0018200000..0x00182fffff -> 0x0000000000
[    0.066169] qcom-pcie 18000000.pcie:      MEM 0x0018300000..0x001fffffff -> 0x0018300000
[    0.066446] qcom-pcie 20000000.pcie: host bridge /soc@0/pcie@20000000 ranges:
[    0.066476] qcom-pcie 20000000.pcie:       IO 0x0020200000..0x00202fffff -> 0x0000000000
[    0.066496] qcom-pcie 20000000.pcie:      MEM 0x0020300000..0x0027ffffff -> 0x0020300000
[    0.074513] bam-dma-engine 704000.dma-controller: num-channels unspecified in dt
[    0.074525] bam-dma-engine 704000.dma-controller: num-ees unspecified in dt
[    0.077681] Serial: 8250/16550 driver, 2 ports, IRQ sharing disabled
[    0.078315] msm_serial 78b1000.serial: msm_serial: detected port #0
[    0.078339] msm_serial 78b1000.serial: uartclk = 1843199
[    0.078557] 78b1000.serial: ttyMSM0 at MMIO 0x78b1000 (irq = 21, base_baud = 115199) is a MSM
[    0.078580] msm_serial: console setup on port #0
[    0.078625] printk: console [ttyMSM0] enabled
[    0.196727] qcom-pcie 20000000.pcie: iATU: unroll T, 8 ob, 8 ib, align 4K, limit 1024G
[    0.196727] qcom-pcie 18000000.pcie: iATU: unroll T, 8 ob, 8 ib, align 4K, limit 1024G
[    0.197270] msm_serial: driver initialized
[    0.202311] qcom-pcie 10000000.pcie: iATU: unroll T, 8 ob, 8 ib, align 4K, limit 1024G
[    0.915040] loop: module loaded
[    0.921476] spi-nand spi0.0: GigaDevice SPI NAND was found.
[    0.921965] spi-nand spi0.0: 256 MiB, block size: 128 KiB, page size: 2048, OOB size: 64
[    1.194769] qcom-pcie 18000000.pcie: Phy link never came up
[    1.194786] qcom-pcie 20000000.pcie: Phy link never came up
[    1.195035] qcom-pcie 18000000.pcie: PCI host bridge to bus 0003:00
[    1.199219] qcom-pcie 20000000.pcie: PCI host bridge to bus 0002:00
[    1.204700] pci_bus 0003:00: root bus resource [bus 00-ff]
[    1.210964] pci_bus 0002:00: root bus resource [bus 00-ff]
[    1.217204] pci_bus 0003:00: root bus resource [io  0x100000-0x1fffff] (bus address [0x0000-0xfffff])
[    1.222755] pci_bus 0002:00: root bus resource [io  0x200000-0x2fffff] (bus address [0x0000-0xfffff])
[    1.228230] pci_bus 0003:00: root bus resource [mem 0x18300000-0x1fffffff]
[    1.237517] pci_bus 0002:00: root bus resource [mem 0x20300000-0x27ffffff]
[    1.246740] pci 0003:00:00.0: [17cb:1108] type 01 class 0x060400
[    1.253499] pci 0002:00:00.0: [17cb:1108] type 01 class 0x060400
[    1.260353] pci 0003:00:00.0: reg 0x10: [mem 0x00000000-0x00000fff]
[    1.266522] pci 0002:00:00.0: reg 0x10: [mem 0x00000000-0x00000fff]
[    1.272554] pci 0003:00:00.0: PME# supported from D0 D3hot D3cold
[    1.278541] pci 0002:00:00.0: PME# supported from D0 D3hot D3cold
[    1.286861] pci 0003:00:00.0: BAR 0: assigned [mem 0x18300000-0x18300fff]
[    1.293056] pci 0002:00:00.0: BAR 0: assigned [mem 0x20300000-0x20300fff]
[    1.297073] pci 0003:00:00.0: PCI bridge to [bus 01-ff]
[    1.303833] pci 0002:00:00.0: PCI bridge to [bus 01-ff]
[    1.310609] pci_bus 0003:00: resource 4 [io  0x100000-0x1fffff]
[    1.315642] pci_bus 0002:00: resource 4 [io  0x200000-0x2fffff]
[    1.320842] pci_bus 0003:00: resource 5 [mem 0x18300000-0x1fffffff]
[    1.326749] pci_bus 0002:00: resource 5 [mem 0x20300000-0x27ffffff]
[    1.334368] pcieport 0003:00:00.0: PME: Signaling with IRQ 39
[    1.340546] pcieport 0002:00:00.0: PME: Signaling with IRQ 41
[    1.345412] pcieport 0003:00:00.0: AER: enabled with IRQ 39
[    1.351240] pcieport 0002:00:00.0: AER: enabled with IRQ 41
[    1.894794] qcom-pcie 10000000.pcie: Phy link never came up
[    1.894980] qcom-pcie 10000000.pcie: PCI host bridge to bus 0001:00
[    1.899170] pci_bus 0001:00: root bus resource [bus 00-ff]
[    1.905441] pci_bus 0001:00: root bus resource [io  0x0000-0xfffff]
[    1.910973] pci_bus 0001:00: root bus resource [mem 0x10300000-0x17ffffff]
[    1.917168] pci 0001:00:00.0: [17cb:1108] type 01 class 0x060400
[    1.924088] pci 0001:00:00.0: reg 0x10: [mem 0x00000000-0x00000fff]
[    1.930308] pci 0001:00:00.0: PME# supported from D0 D3hot D3cold
[    1.938395] pci 0001:00:00.0: BAR 0: assigned [mem 0x10300000-0x10300fff]
[    1.942489] pci 0001:00:00.0: PCI bridge to [bus 01-ff]
[    1.949270] pci_bus 0001:00: resource 4 [io  0x0000-0xfffff]
[    1.954288] pci_bus 0001:00: resource 5 [mem 0x10300000-0x17ffffff]
[    1.961923] pcieport 0001:00:00.0: PME: Signaling with IRQ 43
[    1.966381] pcieport 0001:00:00.0: AER: enabled with IRQ 43
[    2.148606] 4 fixed-partitions partitions found on MTD device spi0.0
[    2.148640] Creating 4 MTD partitions on "spi0.0":
[    2.154021] 0x000000000000-0x000000080000 : "0:training"
[    2.159663] 0x000000080000-0x0000000c0000 : "0:license"
[    2.164697] 0x0000000c0000-0x000003cc0000 : "rootfs"
[    2.237101] mtd: setting mtd2 (rootfs) as root device
[    2.237551] mtdsplit: no squashfs found in "rootfs"
[    2.241135] 0x000003cc0000-0x0000078c0000 : "rootfs_1"
[    2.315135] spi_qup 78b5000.spi: IN:block:16, fifo:64, OUT:block:16, fifo:64
[    2.315723] spi-nor spi1.0: mx25u25635f (32768 Kbytes)
[    2.326847] 21 fixed-partitions partitions found on MTD device spi1.0
[    2.327119] Creating 21 MTD partitions on "spi1.0":
[    2.332704] 0x000000000000-0x0000000c0000 : "0:sbl1"
[    2.337769] 0x0000000c0000-0x0000000d0000 : "0:mibib"
[    2.342907] 0x0000000d0000-0x0000000f0000 : "0:bootconfig"
[    2.347881] 0x0000000f0000-0x000000110000 : "0:bootconfig1"
[    2.353222] 0x000000110000-0x000000290000 : "0:qsee"
[    2.358709] 0x000000290000-0x000000410000 : "0:qsee_1"
[    2.363903] 0x000000410000-0x000000420000 : "0:devcfg"
[    2.368868] 0x000000420000-0x000000430000 : "0:devcfg_1"
[    2.373978] 0x000000430000-0x000000440000 : "0:apdp"
[    2.379479] 0x000000440000-0x000000450000 : "0:apdp_1"
[    2.384421] 0x000000450000-0x000000490000 : "0:tme"
[    2.389376] 0x000000490000-0x0000004d0000 : "0:tme_1"
[    2.394120] 0x0000004d0000-0x0000004f0000 : "0:rpm"
[    2.399367] 0x0000004f0000-0x000000510000 : "0:rpm_1"
[    2.404018] 0x000000510000-0x000000520000 : "0:cdt"
[    2.409371] 0x000000520000-0x000000530000 : "0:cdt_1"
[    2.413927] 0x000000530000-0x000000540000 : "0:appsblenv"
[    2.419085] 0x000000540000-0x0000005e0000 : "0:appsbl"
[    2.424506] 0x0000005e0000-0x000000680000 : "0:appsbl_1"
[    2.429570] 0x000000680000-0x000000780000 : "0:art"
[    2.435067] 0x000000780000-0x000000800000 : "0:ethphyfw"
[    2.826178] dummy# (dummy): netif_napi_add_weight() called with weight 128
[    2.827790] qcom_ppe 3a000000.ethernet: EDMA configuration successful
[    2.832072] qcom_ppe 3a000000.ethernet lan1 (uninitialized): GMAC1 Using random MAC address - 06:2e:53:81:a9:5b
[    3.035402] qcom_ppe 3a000000.ethernet lan1 (uninitialized): PHY [90000.mdio-1:10] driver [Qualcomm QCA8075] (irq=POLL)
[    3.036084] qcom_ppe 3a000000.ethernet lan2 (uninitialized): GMAC2 Using random MAC address - de:3a:9d:44:e3:e1
[    3.135029] qcom_ppe 3a000000.ethernet lan2 (uninitialized): PHY [90000.mdio-1:11] driver [Qualcomm QCA8075] (irq=POLL)
[    3.135687] qcom_ppe 3a000000.ethernet lan3 (uninitialized): GMAC3 Using random MAC address - de:9a:96:09:c8:22
[    3.225017] qcom_ppe 3a000000.ethernet lan3 (uninitialized): PHY [90000.mdio-1:12] driver [Qualcomm QCA8075] (irq=POLL)
[    3.225624] qcom_ppe 3a000000.ethernet lan4 (uninitialized): GMAC4 Using random MAC address - 4e:c5:d4:a4:77:af
[    3.305031] qcom_ppe 3a000000.ethernet lan4 (uninitialized): PHY [90000.mdio-1:13] driver [Qualcomm QCA8075] (irq=POLL)
[    3.305645] qcom_ppe 3a000000.ethernet lan5 (uninitialized): GMAC5 Using random MAC address - 12:bf:4a:53:d6:72
[    3.325220] qcom_ppe 3a000000.ethernet lan5 (uninitialized): PHY [90000.mdio-1:08] driver [Aquantia AQR113C] (irq=POLL)
[    3.325818] qcom_ppe 3a000000.ethernet wan (uninitialized): GMAC6 Using random MAC address - 4a:eb:e3:dc:a2:eb
[    3.346033] qcom_ppe 3a000000.ethernet wan (uninitialized): PHY [90000.mdio-1:00] driver [Aquantia AQR113C] (irq=POLL)
[    3.347119] i2c_dev: i2c /dev entries driver
[    3.360344] sdhci: Secure Digital Host Controller Interface driver
[    3.360613] sdhci: Copyright(c) Pierre Ossman
[    3.366621] sdhci-pltfm: SDHCI platform and OF driver helper
[    3.388911] NET: Registered PF_INET6 protocol family
[    3.390348] Segment Routing with IPv6
[    3.392959] In-situ OAM (IOAM) with IPv6
[    3.396575] NET: Registered PF_PACKET protocol family
[    3.400654] 8021q: 802.1Q VLAN Support v1.8
[    3.434724] cpufreq: cpufreq_online: CPU0: Running at unlisted initial frequency: 24000 KHz, changing to: 936000 KHz
[    3.435587] ubi0: attaching mtd2
[    3.801888] ubi0: scanning is finished
[    3.810812] ubi0: attached mtd2 (name "rootfs", size 60 MiB)
[    3.810837] ubi0: PEB size: 131072 bytes (128 KiB), LEB size: 126976 bytes
[    3.815542] ubi0: min./max. I/O unit sizes: 2048/2048, sub-page size 2048
[    3.822212] ubi0: VID header offset: 2048 (aligned 2048), data offset: 4096
[    3.829078] ubi0: good PEBs: 480, bad PEBs: 0, corrupted PEBs: 0
[    3.835843] ubi0: user volume: 3, internal volumes: 1, max. volumes count: 128
[    3.842090] ubi0: max/mean erase counter: 3/1, WL threshold: 4096, image sequence number: 1695906269
[    3.849129] ubi0: available PEBs: 0, total reserved PEBs: 480, PEBs reserved for bad PEB handling: 40
[    3.858423] ubi0: background thread "ubi_bgt0d" started, PID▒[    3.875940] Freeing unused kernel memory: 7616K
[    3.876057] Run /init as init process
[    3.954194] init: Console is alive
[    3.954264] init: - watchdog -
[    3.961449] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    3.964970] gpio_button_hotplug: loading out-of-tree module taints kernel.
[    3.972527] ------------[ cut here ]------------
[    3.973975] nss_cc_nssnoc_nss_csr_clk status stuck at 'on'
[    3.973988] WARNING: CPU: 0 PID: 819 at drivers/clk/qcom/clk-branch.c:86 clk_branch_wait+0x114/0x124
[    3.984049] Modules linked in: dwc3(+) dwc3_qcom gpio_button_hotplug(O) vfat fat crc32c_generic
[    3.993337] CPU: 0 PID: 819 Comm: kmodloader Tainted: G           O       6.6.71 #0
[    4.001755] Hardware name: Qualcomm Technologies, Inc. IPQ9574/AP-AL02-C7 (DT)
[    4.009394] pstate: 604000c5 (nZCv daIF +PAN -UAO -TCO -DIT -SSBS BTYPE=--)
[    4.016685] pc : clk_branch_wait+0x114/0x124
[    4.023540] lr : clk_branch_wait+0x114/0x124
[    4.028053] sp : ffffffc0842db790
[    4.032306] x29: ffffffc0842db790 x28: 0000000000000001 x27: ffffffc0793fc2b0
[    4.035522] x26: ffffffc0793fc240 x25: ffffffc0793fc118 x24: 0000000000000003
[    4.042640] x23: ffffffc080a62f00 x22: 0000000000000000 x21: ffffffc08044adc4
[    4.049758] x20: 0000000000000000 x19: ffffffc08131b480 x18: 0000000000000102
[    4.056875] x17: 0000000000000000 x16: 0000000000000000 x15: ffffffc08128bdf0
[    4.063995] x14: 0000000000000306 x13: 0000000000000102 x12: 00000000ffffffea
[    4.071112] x11: 00000000ffffefff x10: ffffffc0812e3df0 x9 : ffffffc08128bd98
[    4.078230] x8 : 0000000000017fe8 x7 : c0000000ffffefff x6 : 0000000000057fa8
[    4.085348] x5 : 0000000000000fff x4 : 0000000000000000 x3 : ffffffc0842db590
[    4.092466] x2 : ffffffc08128bd30 x1 : ffffffc08128bd30 x0 : 000000000000002e
[    4.099584] Call trace:
[    4.106699]  clk_branch_wait+0x114/0x124
[    4.108957]  clk_branch2_disable+0x2c/0x3c
[    4.113123]  clk_core_disable+0x60/0xac
[    4.117029]  clk_disable+0x30/0x4c
[    4.120761]  icc_clk_set+0xc0/0xdc
[    4.124234]  icc_sync_state+0xc8/0x10c
[    4.127619]  device_links_flush_sync_list+0x8c/0x11c
[    4.131353]  device_links_driver_bound+0x1f4/0x27c
[    4.136474]  driver_bound+0x74/0xb8
[    4.141074]  really_probe+0x1e0/0x2b0
[    4.144459]  __driver_probe_device+0x78/0x128
[    4.148280]  driver_probe_device+0x40/0xdc
[    4.152620]  __driver_attach+0x90/0x160
[    4.156613]  bus_for_each_dev+0x64/0xa4
[    4.160345]  driver_attach+0x24/0x30
[    4.164164]  bus_add_driver+0xe4/0x208
[    4.167983]  driver_register+0x5c/0x124
[    4.171543]  __platform_driver_register+0x28/0x34
[    4.175276]  init_module+0x20/0x1000 [dwc3]
[    4.180137]  do_one_initcall+0x6c/0x1fc
[    4.184130]  do_init_module+0x58/0x1d8
[    4.187949]  load_module+0x18ac/0x1940
[    4.191769]  __do_sys_init_module+0x218/0x264
[    4.195501]  __arm64_sys_init_module+0x1c/0x28
[    4.199929]  invoke_syscall.constprop.0+0x5c/0x108
[    4.204270]  do_el0_svc+0x40/0xc8
[    4.209042]  el0_svc+0x28/0x9c
[    4.212427]  el0t_64_sync_handler+0x120/0x12c
[    4.215380]  el0t_64_sync+0x178/0x17c
[    4.219807] ---[ end trace 0000000000000000 ]---
[    4.223771] ------------[ cut here ]------------
[    4.228141] nss_cc_nssnoc_imem_qsb_clk status stuck at 'on'
[    4.228154] WARNING: CPU: 2 PID: 819 at drivers/clk/qcom/clk-branch.c:86 clk_branch_wait+0x114/0x124
[    4.238041] Modules linked in: dwc3(+) dwc3_qcom gpio_button_hotplug(O) vfat fat crc32c_generic
[    4.247416] CPU: 2 PID: 819 Comm: kmodloader Tainted: G        W  O       6.6.71 #0
[    4.255835] Hardware name: Qualcomm Technologies, Inc. IPQ9574/AP-AL02-C7 (DT)
[    4.263473] pstate: 604000c5 (nZCv daIF +PAN -UAO -TCO -DIT -SSBS BTYPE=--)
[    4.270764] pc : clk_branch_wait+0x114/0x124
[    4.277620] lr : clk_branch_wait+0x114/0x124
[    4.282134] sp : ffffffc0842db790
[    4.286386] x29: ffffffc0842db790 x28: 0000000000000001 x27: ffffffc0793fc2b0
[    4.289601] x26: ffffffc0793fc240 x25: ffffffc0793fc118 x24: 0000000000000003
[    4.296720] x23: ffffffc080a62f20 x22: 0000000000000000 x21: ffffffc08044adc4
[    4.303838] x20: 0000000000000000 x19: ffffffc08131b4c0 x18: 0000000000000135
[    4.310955] x17: 0000000000000000 x16: 0000000000000000 x15: ffffffc08128bdf0
[    4.318074] x14: 000000000000039f x13: 0000000000000135 x12: 00000000ffffffea
[    4.325192] x11: 00000000ffffefff x10: ffffffc0812e3df0 x9 : ffffffc08128bd98
[    4.332309] x8 : 0000000000017fe8 x7 : c0000000ffffefff x6 : 0000000000057fa8
[    4.339428] x5 : 0000000000000fff x4 : 0000000000000000 x3 : ffffffc0842db590
[    4.346547] x2 : ffffffc08128bd30 x1 : ffffffc08128bd30 x0 : 000000000000002f
[    4.353665] Call trace:
[    4.360778]  clk_branch_wait+0x114/0x124
[    4.363036]  clk_branch2_disable+0x2c/0x3c
[    4.367204]  clk_core_disable+0x60/0xac
[    4.371109]  clk_disable+0x30/0x4c
[    4.374840]  icc_clk_set+0xc0/0xdc
[    4.378313]  icc_sync_state+0xc8/0x10c
[    4.381699]  device_links_flush_sync_list+0x8c/0x11c
[    4.385433]  device_links_driver_bound+0x1f4/0x27c
[    4.390554]  driver_bound+0x74/0xb8
[    4.395154]  really_probe+0x1e0/0x2b0
[    4.398539]  __driver_probe_device+0x78/0x128
[    4.402360]  driver_probe_device+0x40/0xdc
[    4.406699]  __driver_attach+0x90/0x160
[    4.410692]  bus_for_each_dev+0x64/0xa4
[    4.414424]  driver_attach+0x24/0x30
[    4.418244]  bus_add_driver+0xe4/0x208
[    4.422063]  driver_register+0x5c/0x124
[    4.425622]  __platform_driver_register+0x28/0x34
[    4.429356]  init_module+0x20/0x1000 [dwc3]
[    4.434217]  do_one_initcall+0x6c/0x1fc
[    4.438210]  do_init_module+0x58/0x1d8
[    4.442029]  load_module+0x18ac/0x1940
[    4.445848]  __do_sys_init_module+0x218/0x264
[    4.449582]  __arm64_sys_init_module+0x1c/0x28
[    4.454009]  invoke_syscall.constprop.0+0x5c/0x108
[    4.458349]  do_el0_svc+0x40/0xc8
[    4.463122]  el0_svc+0x28/0x9c
[    4.466507]  el0t_64_sync_handler+0x120/0x12c
[    4.469459]  el0t_64_sync+0x178/0x17c
[    4.473886] ---[ end trace 0000000000000000 ]---
[    4.477828] ------------[ cut here ]------------
[    4.482221] nss_cc_nssnoc_imem_ahb_clk status stuck at 'on'
[    4.482230] WARNING: CPU: 2 PID: 819 at drivers/clk/qcom/clk-branch.c:86 clk_branch_wait+0x114/0x124
[    4.492121] Modules linked in: dwc3(+) dwc3_qcom gpio_button_hotplug(O) vfat fat crc32c_generic
[    4.501496] CPU: 2 PID: 819 Comm: kmodloader Tainted: G        W  O       6.6.71 #0
[    4.509915] Hardware name: Qualcomm Technologies, Inc. IPQ9574/AP-AL02-C7 (DT)
[    4.517553] pstate: 604000c5 (nZCv daIF +PAN -UAO -TCO -DIT -SSBS BTYPE=--)
[    4.524845] pc : clk_branch_wait+0x114/0x124
[    4.531700] lr : clk_branch_wait+0x114/0x124
[    4.536214] sp : ffffffc0842db790
[    4.540466] x29: ffffffc0842db790 x28: 0000000000000001 x27: ffffffc0793fc2b0
[    4.543682] x26: ffffffc0793fc240 x25: ffffffc0793fc118 x24: 0000000000000003
[    4.550799] x23: ffffffc080a62f40 x22: 0000000000000000 x21: ffffffc08044adc4
[    4.557918] x20: 0000000000000000 x19: ffffffc08131b500 x18: 0000000000000168
[    4.565036] x17: 0000000000000000 x16: 0000000000000000 x15: ffffffc08128bdf0
[    4.572154] x14: 0000000000000438 x13: 0000000000000168 x12: 00000000ffffffea
[    4.579271] x11: 00000000ffffefff x10: ffffffc0812e3df0 x9 : ffffffc08128bd98
[    4.586390] x8 : 0000000000017fe8 x7 : c0000000ffffefff x6 : 0000000000057fa8
[    4.593507] x5 : 0000000000000fff x4 : 0000000000000000 x3 : ffffffc0842db590
[    4.600626] x2 : ffffffc08128bd30 x1 : ffffffc08128bd30 x0 : 000000000000002f
[    4.607744] Call trace:
[    4.614857]  clk_branch_wait+0x114/0x124
[    4.617116]  clk_branch2_disable+0x2c/0x3c
[    4.621282]  clk_core_disable+0x60/0xac
[    4.625189]  clk_disable+0x30/0x4c
[    4.628921]  icc_clk_set+0xc0/0xdc
[    4.632392]  icc_sync_state+0xc8/0x10c
[    4.635779]  device_links_flush_sync_list+0x8c/0x11c
[    4.639513]  device_links_driver_bound+0x1f4/0x27c
[    4.644634]  driver_bound+0x74/0xb8
[    4.649234]  really_probe+0x1e0/0x2b0
[    4.652619]  __driver_probe_device+0x78/0x128
[    4.656439]  driver_probe_device+0x40/0xdc
[    4.660780]  __driver_attach+0x90/0x160
[    4.664771]  bus_for_each_dev+0x64/0xa4
[    4.668504]  driver_attach+0x24/0x30
[    4.672324]  bus_add_driver+0xe4/0x208
[    4.676144]  driver_register+0x5c/0x124
[    4.679702]  __platform_driver_register+0x28/0x34
[    4.683437]  init_module+0x20/0x1000 [dwc3]
[    4.688296]  do_one_initcall+0x6c/0x1fc
[    4.692289]  do_init_module+0x58/0x1d8
[    4.696108]  load_module+0x18ac/0x1940
[    4.699928]  __do_sys_init_module+0x218/0x264
[    4.703661]  __arm64_sys_init_module+0x1c/0x28
[    4.708088]  invoke_syscall.constprop.0+0x5c/0x108
[    4.712430]  do_el0_svc+0x40/0xc8
[    4.717203]  el0_svc+0x28/0x9c
[    4.720587]  el0t_64_sync_handler+0x120/0x12c
[    4.723540]  el0t_64_sync+0x178/0x17c
[    4.727967] ---[ end trace 0000000000000000 ]---
[    4.733707] xhci-hcd xhci-hcd.1.auto: xHCI Host Controller
[    4.736367] xhci-hcd xhci-hcd.1.auto: new USB bus registered, assigned bus number 1
[    4.742440] xhci-hcd xhci-hcd.1.auto: hcc params 0x0220fe65 hci version 0x110 quirks 0x0000008000000010
[    4.749177] xhci-hcd xhci-hcd.1.auto: irq 76, io mem 0x08a00000
[    4.758581] xhci-hcd xhci-hcd.1.auto: xHCI Host Controller
[    4.764428] xhci-hcd xhci-hcd.1.auto: new USB bus registered, assigned bus number 2
[    4.769989] xhci-hcd xhci-hcd.1.auto: Host supports USB 3.0 SuperSpeed
[    4.777771] hub 1-0:1.0: USB hub found
[    4.784143] hub 1-0:1.0: 1 port detected
[    4.788030] usb usb2: We don't know the algorithms for LPM for this host, disabling LPM.
[    4.792135] hub 2-0:1.0: USB hub found
[    4.800039] hub 2-0:1.0: 1 port detected
[    4.804783] usbcore: registered new interface driver usb-storage
[    4.808034] usbcore: registered new interface driver uas
[    4.813718] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    4.822243] init: - preinit -
[    8.144740] random: crng init done
[    8.195534] qcom_ppe 3a000000.ethernet lan1: configuring for inband/qsgmii link mode
Press the [f] key and hit [enter] to enter failsafe mode
Press the [1], [2], [3] or [4] key and hit [enter] to select the debug level
[   12.477266] procd: - early -
[   12.477320] procd: - watchdog -
[   12.995858] procd: - watchdog -
[   12.996025] procd: - ubus -
[   13.049926] procd: - init -
Please press Enter to activate this console.
[   13.126265] kmodloader: loading kernel modules from /etc/modules.d/*
[   13.176939] usbcore: registered new interface driver ums-alauda
[   13.177159] usbcore: registered new interface driver ums-cypress
[   13.181857] usbcore: registered new interface driver ums-datafab
[   13.188165] usbcore: registered new interface driver ums-freecom
[   13.194112] usbcore: registered new interface driver ums-isd200
[   13.200507] usbcore: registered new interface driver ums-jumpshot
[   13.205769] usbcore: registered new interface driver ums-karma
[   13.212048] usbcore: registered new interface driver ums-sddr09
[   13.217275] urngd: v1.0.2 started.
[   13.223402] usbcore: registered new interface driver ums-sddr55
[   13.227088] usbcore: registered new interface driver ums-usbat
[   13.240286] PPP generic driver version 2.4.2
[   13.240568] NET: Registered PF_PPPOX protocol family
[   13.244540] kmodloader: done loading kernel modules from /etc/modules.d/*
[   14.703779] mtdblock: MTD device 'rootfs' is NAND, please consider using UBI block devices instead.
[   16.890814] qcom_ppe 3a000000.ethernet lan1: configuring for inband/qsgmii link mode
[   17.055178] br-lan: port 1(lan1) entered blocking state
[   17.055210] br-lan: port 1(lan1) entered disabled state
[   17.059222] qcom_ppe 3a000000.ethernet lan1: entered allmulticast mode
[   17.064540] qcom_ppe 3a000000.ethernet lan1: entered promiscuous mode
[   17.075716] qcom_ppe 3a000000.ethernet lan2: configuring for inband/qsgmii link mode
[   17.245234] br-lan: port 2(lan2) entered blocking state
[   17.245265] br-lan: port 2(lan2) entered disabled state
[   17.249279] qcom_ppe 3a000000.ethernet lan2: entered allmulticast mode
[   17.254649] qcom_ppe 3a000000.ethernet lan2: entered promiscuous mode
[   17.263430] qcom_ppe 3a000000.ethernet lan3: configuring for inband/qsgmii link mode
[   17.435126] br-lan: port 3(lan3) entered blocking state
[   17.435152] br-lan: port 3(lan3) entered disabled state
[   17.439167] qcom_ppe 3a000000.ethernet lan3: entered allmulticast mode
[   17.444524] qcom_ppe 3a000000.ethernet lan3: entered promiscuous mode
[   17.453303] qcom_ppe 3a000000.ethernet lan4: configuring for inband/qsgmii link mode
[   17.625110] br-lan: port 4(lan4) entered blocking state
[   17.625135] br-lan: port 4(lan4) entered disabled state
[   17.629150] qcom_ppe 3a000000.ethernet lan4: entered allmulticast mode
[   17.634582] qcom_ppe 3a000000.ethernet lan4: entered promiscuous mode
[   17.643478] qcom_ppe 3a000000.ethernet lan5: configuring for inband/usxgmii link mode
[   17.675155] br-lan: port 5(lan5) entered blocking state
[   17.675180] br-lan: port 5(lan5) entered disabled state
[   17.679194] qcom_ppe 3a000000.ethernet lan5: entered allmulticast mode
[   17.684560] qcom_ppe 3a000000.ethernet lan5: entered promiscuous mode
[   17.694068] qcom_ppe 3a000000.ethernet wan: configuring for inband/usxgmii link mode
[   19.374892] qcom_ppe 3a000000.ethernet lan2: Link is Up - 1Gbps/Full - flow control rx/tx
[   19.374912] br-lan: port 2(lan2) entered blocking state
[   19.382047] br-lan: port 2(lan2) entered forwarding state



BusyBox v1.36.1 (2024-12-13 11:20:36 UTC) built-in shell (ash)

  _______                     ________        __
 |       |.-----.-----.-----.|  |  |  |.----.|  |_
 |   -   ||  _  |  -__|     ||  |  |  ||   _||   _|
 |_______||   __|_____|__|__||________||__|  |____|
          |__| W I R E L E S S   F R E E D O M
 -----------------------------------------------------
 OpenWrt SNAPSHOT, r0+28124-fc4b345ee3
 -----------------------------------------------------
=== WARNING! =====================================
There is no root password defined on this device!
Use the "passwd" command to set up a new password
in order to prevent unauthorized SSH logins.
--------------------------------------------------
root@OpenWrt:~# ping 192.168.1.10
PING 192.168.1.10 (192.168.1.10): 56 data bytes
64 bytes from 192.168.1.10: seq=0 ttl=128 time=0.553 ms
64 bytes from 192.168.1.10: seq=1 ttl=128 time=0.544 ms
64 bytes from 192.168.1.10: seq=2 ttl=128 time=0.504 ms
64 bytes from 192.168.1.10: seq=3 ttl=128 time=0.467 ms
64 bytes from 192.168.1.10: seq=4 ttl=128 time=0.522 ms
^C
--- 192.168.1.10 ping statistics ---
5 packets transmitted, 5 packets received, 0% packet loss
round-trip min/avg/max = 0.467/0.518/0.553 ms
root@OpenWrt:~# iperf3 -s
-----------------------------------------------------------
Server listening on 5201 (test #1)
-----------------------------------------------------------
Accepted connection from 192.168.1.10, port 57664
[  5] local 192.168.1.1 port 5201 connected to 192.168.1.10 port 57665
[ ID] Interval           Transfer     Bitrate
[  5]   0.00-1.00   sec  38.9 MBytes   326 Mbits/sec
[  5]   1.00-2.00   sec  39.1 MBytes   328 Mbits/sec
[  5]   2.00-3.00   sec  39.0 MBytes   327 Mbits/sec
[  5]   3.00-4.00   sec  39.1 MBytes   328 Mbits/sec
[  5]   4.00-5.00   sec  39.1 MBytes   328 Mbits/sec
[  5]   5.00-6.00   sec  38.8 MBytes   325 Mbits/sec
[  5]   6.00-7.00   sec  39.1 MBytes   328 Mbits/sec
[  5]   7.00-8.00   sec  39.1 MBytes   328 Mbits/sec
[  5]   8.00-9.00   sec  39.0 MBytes   327 Mbits/sec
[  5]   9.00-10.00  sec  39.1 MBytes   328 Mbits/sec
[  5]  10.00-10.02  sec   896 KBytes   325 Mbits/sec
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID] Interval           Transfer     Bitrate
[  5]   0.00-10.02  sec   391 MBytes   327 Mbits/sec                  receiver
-----------------------------------------------------------
Server listening on 5201 (test #2)
-----------------------------------------------------------
^Ciperf3: interrupt - the server has terminated
root@OpenWrt:~#
```
(ignore the clock warning, they will be fixed later... Also the low speed is something that is getting handled...)

Next step... ath12K support...

Ansuel committed:
kernel: usb: add qualcommbe to the supported target of dwc3-qcom

Add qualcommbe to the supported target of dwc3-qcom kernel module. USB3
is correctly supported on IPQ95xx and can be enabled.

Link: https://github.com/openwrt/openwrt/pull/17788
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
qualcommbe: ipq95xx: Add USB3 kmods and drop NSS-DP unneeded package

Add USB3 kmods to the default package list, USB3 correctly works on
IPQ95xx hence it can be enabled.

While at it drop the NSS-DP as it was added by mistake as the ethrnet
platform is handled differently on this SoC with pending upstream
drivers.

Link: https://github.com/openwrt/openwrt/pull/17788
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
qualcommbe: ipq95xx: Add PCIe upstream patch and related nodes

Add PCIe upstream patch and related nodes to enable PCIe on IPQ95xx.

Minimal change were required to backport the patch and apply on current
kernel. Refresh all affected patch.

Link: https://github.com/openwrt/openwrt/pull/17788
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
qualcommbe: ipq95xx: Refresh dts SPI-NAND patch to v14

Refresh dts SPI-NAND patch to to v14. This is to keep stuff synced with
current pending patch revision and make it easier to replace patch
later (and discover something broke in the meantime)

Link: https://github.com/openwrt/openwrt/pull/17788
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
qualcommbe: ipq95xx: Define default partition table for RFB SPI-NAND

Define default partition table for SPI-NAND mounted on reference board.
This is where is normally placed the rootfs UBI.

Link: https://github.com/openwrt/openwrt/pull/17788
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
qualcommbe: ipq95xx: Drop redundant comment in NSSCC clock patch

Drop redundant comment in NSSCC clock patch. The problem has been
identified hence the comment doesn't apply anymore.

Link: https://github.com/openwrt/openwrt/pull/17788
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
qualcommbe: ipq95xx: Refresh the NSSCC and PORT patch for new PCIe patches

Refresh the NSSCC patch for new PCIe patches. To keep track of fuzz
changes for the IPQ95xx patches, patch are not refreshed currently.

For the specific case of NSSCC patch, quilt gets confused and apply the
patch in the wrong node, putting it in the RPM node (causing all kind of
funny errors at runtime)

Correctly fix the patch to put the node right after the PCIe nodes.

Also the PORT patch need to be refreshed as the gpio header is added by
the PCIe patch.

Link: https://github.com/openwrt/openwrt/pull/17788
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
qualcommbe: ipq95xx: Add pending patch fixing NSSCC boot stall

Add pending patch fixing NSSCC boot stall. These patch are needed to
prevent the ICC to disable critical clock for NSSCC NOC.

Without these the system will stall and reboot with watchdog.

While at it also remove an extra clock from DTSI as it currently have no
use. Original patch is not modified to keep consistency with series
proposed upstream.

Link: https://github.com/openwrt/openwrt/pull/17788
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
qualcommbe: ipq95xx: Make RPM partition read-only

Make the RPM partition read-only. This was a mistake and a leftover from
staging branch but I can take this mistake as an excuse to document the
current problem with RPM.

It might happen that a board ship with a broken RPM .mbn, broken not in
the sense that the board doesn't boot or it's a brick but broken in the
sense that it's outdaed and suffer from a bug fixed in new version.

This bug consist in a problem with the regulators between USB and NSS.
The old RPM mess with the NSS regulator (l2) and change the voltage for
it while configuring the USB regulator (l5).
This cause the ethernet subsystem to malfunction with the port not
working.

To workaround this, it's needed to disable RPM handling and CPUFreq.
With these 2 disabled, the old RPM doesn't touch regulators and Ethernet
works correctly.

New RPM correctly handle regulators for USB (l5) and doesn't suffer from
this problem. A solution for this is getting discussed with QCOM hoping
to get some good feedback for it.

Link: https://github.com/openwrt/openwrt/pull/17788
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

mantas-p commented on target/linux/qualcommbe/patches-6.6/023-01-v6.14-arm64-dts-qcom-ipq9574-Add-PCIe-PHYs-and-controller-.patch:
```diff
@@ -0,0 +1,473 @@
++				     <GIC_SPI 137 IRQ_TYPE_LEVEL_HIGH>,
++				     <GIC_SPI 141 IRQ_TYPE_LEVEL_HIGH>,
++				     <GIC_SPI 142 IRQ_TYPE_LEVEL_HIGH>,
++				     <GIC_SPI 143 IRQ_TYPE_LEVEL_HIGH>;
```
These MSI lines are not correct [1]. Radio won't actually start-up in this slot. I use the following workaround:

``` diff
 +                                    "axi_m",
 +                                    "aux",
 +                                    "ahb";
-+
++                      msi-parent = <&v2m0>;
 +                      phys = <&pcie1_phy>;
 +                      phy-names = "pciephy";
 +                      interconnects = <&gcc MASTER_ANOC_PCIE1 &gcc SLAVE_ANOC_PCIE1>,
@@ -374,7 +374,7 @@ index 48dfafea46a7..078e2950acd2 100644
 +                                    "axi_m",
 +                                    "aux",
 +                                    "ahb";
-+
++                      msi-parent = <&v2m0>;
 +                      phys = <&pcie3_phy>;
 +                      phy-names = "pciephy";
 +                      interconnects = <&gcc MASTER_ANOC_PCIE3 &gcc SLAVE_ANOC_PCIE3>,
@@ -454,7 +454,7 @@ index 48dfafea46a7..078e2950acd2 100644
 +                                    "axi_m",
 +                                    "aux",
 +                                    "ahb";
-+
++                      msi-parent = <&v2m0>;
 +                      phys = <&pcie2_phy>;
 +                      phy-names = "pciephy";
 +                      interconnects = <&gcc MASTER_ANOC_PCIE2 &gcc SLAVE_ANOC_PCIE2>,
@@ -533,7 +533,7 @@ index 48dfafea46a7..078e2950acd2 100644
 +                                    "axi_m",
 +                                    "aux",
 +                                    "ahb";
-+
++                      msi-parent = <&v2m0>;
 +                      phys = <&pcie0_phy>;
 +                      phy-names = "pciephy";
 +                      interconnects = <&gcc MASTER_ANOC_PCIE0 &gcc SLAVE_ANOC_PCIE0>,
```

[1] https://patchwork.kernel.org/project/linux-arm-msm/patch/20240801054803.3015572-3-quic_srichara@quicinc.com/#26164605

mantas-p replied:
Earlier revisions had this: https://patchwork.kernel.org/project/linux-arm-msm/patch/20230519090219.15925-5-quic_devipriy@quicinc.com/

Ansuel replied:
Ok will investigate...

hauke commented on target/linux/qualcommbe/patches-6.6/308-02-clk-qcom-nsscc-Attach-required-NSSNOC-clock-to-PM-do.patch:
```diff
@@ -0,0 +1,125 @@
++		.pm = &nsscc_pm_ops,
+ 	},
+ };
+ 
+-- 
+2.47.1
+
```
Please refresh the patches

Ansuel replied:
Ehh i'm keeping the patch this way to catch fuzz

hauke replied:
ok, but the github action should complain about this patch. I do not understand why it does not.

Ansuel replied:
It's source only so it's not compiled. We all base on the get target per script and source only target are not included

Ansuel committed:
generic: move QCOM SPI NAND driver to generic backports

QCOM SPI NAND driver got merged upstream hence we can drop the special
patch from qualcommax and qualcommbe target and move them to the generic
backports directory to reduce patch maintenance.

While at it refresh any affected patch and target and also backport other
minor fixup for the SPI NAND driver merged upstream later.

Link: https://github.com/openwrt/openwrt/pull/17788
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
qualcommbe: disable CONFIG_QCOM_IPA kernel config

CONFIG_QCOM_IPA kernel cofig was enabled by mistake and conflicts with
mac80211 as it indirectly selects QMI HELPERS. Backports project provid
his own version of QMI HELPERS hence it should not be built-in.

Link: https://github.com/openwrt/openwrt/pull/17788
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
